### ISSUE TITLE
Some status messages are no longer shown when sending

### ISSUE THREAD

**Issue by scottkosty** [neomutt/neomutt#2952]
## Expected Behaviour

When sending a message, I used to see "connecting..." "authenticating..." "sending..." "sent" status messages.

## Actual Behaviour

On current master I see only a pause (no status message) and then finally "sending..." and "sent". The "connecting..." and "authenticating..." messages are not shown.

## Steps to Reproduce

This happens whenever I send a message.

## How often does this happen?

- Always

## When did it start to happen?

Bisect leads to 9e94fac5.

## NeoMutt Version

```
NeoMutt 20210205-437-8f4eca
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.11.0-18-generic (x86_64)
ncurses: ncurses 6.2.20201114 (compiled with 6.2.20201114)
libidn: 1.33 (compiled with 1.33)
OpenSSL: OpenSSL 1.1.1j  16 Feb 2021
libnotmuch: 5.3.0
storage: tokyocabinet

Configure options: --ssl --sasl --enable-notmuch --enable-debug --tokyocabinet --prefix=/usr/local

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/local/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls -gpgme -gss +hcache -homespool +idn +inotify 
  -locales_hack -lua +meta -mixmaster +nls +notmuch +openssl +pgp +regex +sasl 
  +smime -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/local/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

**Comment by flatcap** [comment]
Thanks.  That sounds like I've lost a `refresh()` to **force** the screen to be updated.

The curses printing functions, `printw()` etc, cache any changes to the screen until a `refresh()`.

**Comment by flatcap** [comment]
musings...

hmm... odd.  there's a refresh immediately after the logging
theory 2: they're being displayed, but someone's clearing the message window very soon afterwards
I can see the messages in the log (and <show-log-messages>)
and if I run `script` around NeoMutt, I can see the messages were displayed (in 'typescript')

**Comment by flatcap** [comment]
I've pushed a quick fix to [[devel/logging]](https://github.com/neomutt/neomutt/tree/devel/logging).
I need to think a bit more, but this may be the "correct" fix :-)

The info message are now delegated to the MessageWindow.
Setting the text causes the Window to flag itself as needing a WA_RECALC, but that doesn't happen immediately.
Recalc/repaint only happen on `window_redraw()`; the `refresh()` wasn't enough.

**Comment by flatcap** [comment]
More notes to self:

I was worried that `refresh()` would update the screen when I wasn't expecting it.
But... none of the other windows will have updates, they'll only have queued updates (WA_RECALC/WA_REPAINT).
Running `window_redraw()` on a specific window will cause a targetted update only.

**Comment by flatcap** [comment]
Fixed by 7d11a7843

**Comment by scottkosty** [comment]
Tested and works great. Thanks for the quick fix!
